apps/ewallet/test/ewallet/gates/transaction_consumption_consumer_gate_test.exs

```diff
@@ -414,6 +414,36 @@ defmodule EWallet.TransactionConsumptionConsumerGateTest do
          meta do
       initialize_wallet(meta.sender_wallet, 200_000, meta.token)
 
+      {res, consumption} =
+        TransactionConsumptionConsumerGate.consume(meta.sender, %{
+          "formatted_transaction_request_id" => meta.request.id,
+          "idempotency_token" => "123"
+        })
+
+      assert res == :ok
+      assert %TransactionConsumption{} = consumption
+      assert consumption.transaction_request_uuid == meta.request.uuid
+      assert consumption.amount == meta.request.amount
+      assert consumption.wallet_address == meta.sender_wallet.address
+
+      {res, consumption} =
+        TransactionConsumptionConsumerGate.consume(meta.sender, %{
+          "formatted_transaction_request_id" => meta.request.id,
+          "idempotency_token" => "123"
+        })
+
+      assert res == :ok
+      assert %TransactionConsumption{} = consumption
+      assert consumption.transaction_request_uuid == meta.request.uuid
+      assert consumption.amount == meta.request.amount
+      assert consumption.wallet_address == meta.sender_wallet.address
+    end
```

Comment:
Since the test case says "(and is idempotent)", can you also assert that `consumption1.transaction_uuid` and `consumption2.transaction_uuid` is the same to guarantee idempotency?

Also applies to the test case below.